### ISSUE TITLE
Typings broken after 4.2.0

### ISSUE THREAD

**Issue by iCrawl** [winstonjs/winston-daily-rotate-file#246]
After v4.2.0 with the "big" typings change, basically everything broke. Was this even tested?

```
$ rimraf dist && tsc
../../node_modules/winston-daily-rotate-file/index.d.ts:1:8 - error TS1259: Module '"/home/runner/work/***/***/node_modules/winston-transport/index"' can only be default-imported using the 'esModuleInterop' flag

1 import TransportStream from "winston-transport";
         ~~~~~~~~~~~~~~~

  ../../node_modules/winston-transport/index.d.ts:37:1
    37 export = TransportStream;
       ~~~~~~~~~~~~~~~~~~~~~~~~~
    This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.

util/logger.ts:55:7 - error TS2351: This expression is not constructable.
  Type 'typeof import("/home/runner/work/***/***/node_modules/winston-daily-rotate-file/index")' has no construct signatures.

55  	new DailyRotateFile({
    	    ~~~~~~~~~~~~~~~
```

Reading this should be pretty evident whats happening here. So lets go over this, first one? I can't really do anything against, the import is simply wrong because `winston-transport` does not export a default.
https://github.com/winstonjs/winston-daily-rotate-file/blob/21fa56519bf64fbde90baa6ec74c4020b0d08a00/index.d.ts#L1

https://github.com/winstonjs/winston-transport/blob/46db8f3c8cd8b106ade8d7e04a191ee388683d60/index.d.ts#L37

Not everyone uses esModuleInterop so relying on this is quite bad.

Now what about the 2nd one. Not being able to construct DailyRotateFile? But I imported it just fine with `import * as DailyRotateFile from 'winston-daily-rotate-file';` before 4.2.0

Well this is actually incorrect in the typings, they say:
https://github.com/winstonjs/winston-daily-rotate-file/blob/21fa56519bf64fbde90baa6ec74c4020b0d08a00/index.d.ts#L106

But in reality:
https://github.com/winstonjs/winston-daily-rotate-file/blob/21fa56519bf64fbde90baa6ec74c4020b0d08a00/index.js#L6-L7

And even after fixing all of those, I can't go back to the intended way of calling `import * as DailyRotatefile from 'x';` because:
`This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.`

Mainly because of how the typings file is written now.

**Comment by mattberther** [comment]
I do not use typescript and rely on contributions from the community to support the ts declarations. I'll bring @Slessi and @BubbleBear into the conversation here so they can weigh in.

**Comment by Slessi** [comment]
@icrawl ahh poo, I'll fix it later today, sorry

**Comment by mattberther** [comment]
Thanks to @Slessi for the quick fix. I published `winston-daily-rotate-file@4.2.1`.
